Add release notes for july release (#435)
SenseNet/sensenet.github.io: _updates/2020-07-01-july.md

---
title: July 2020
---

## Brick by brick 🧱

> As we get closer and closer to staring [SNaaS](https://docs.sensenet.com/concepts/introduction/04-what-is-snaas), instead of hot new features stabilization and simplification came to the fore. We are working hard on making the core of the admin surface super fast, super effective and easy-to-use. Followings are the updates we've made in the last 4 weeks:

Reviewer: ```suggestion
As we get closer and closer to starting [SNaaS](https://docs.sensenet.com/concepts/introduction/04-what-is-snaas), instead of hot new features stabilization and simplification came to the fore. We are working hard on making the core of the admin surface super fast, super effective and easy-to-use. Followings are the updates we've made in the last 4 weeks:
```

### Admin-ui@1.9.0

- The browseType feature has undergone significant refactoring. From the three browse type only `explore` remained, which means that most of the features can have a common user interface making the work with contents easy-to-learn as you can see the same tree + grid or tree + action ui and with these updates we get a bit closer to a maintainable and a less error prone admin-ui. [#703](https://github.com/SenseNet/sn-client/pull/703)

- Token expiration based browser tab freezing issues are fixed. [#701](https://github.com/SenseNet/sn-client/pull/701)

- Some of the Document Viewer toolbar buttons are now hidden to make it clutter free. Only two zoom options remained and the two of the rotation options are also hidden now. It does not mean that this features are gone from the docviewer component as well, if you use viewer as a standalone component you can customize the toolbar yourself. [#702](https://github.com/SenseNet/sn-client/pull/702)

> - We've refactored the admin-ui drawer items and the undelying logic as well. The built-in drawer items and 'pages' (search, content, users and groups, setup, localization, content types and trash) became fix and cannot be removed or overwritten in personal settings. All of these built-in menu items got a 'protected' root, which means they have a root what you cannot go above being in a scope of a particular menu. The possibility of adding custom drawer items remaind but changed as you can see in the following snippet:

Reviewer 2: ```suggestion
- We've refactored the admin-ui drawer items and the underlying logic as well. The built-in drawer items and 'pages' (search, content, users and groups, setup, localization, content types and trash) became fix and cannot be removed or overwritten in personal settings. All of these built-in menu items got a 'protected' root, which means they have a root what you cannot go above being in a scope of a particular menu. The possibility of adding custom drawer items remained but changed as you can see in the following snippet:
```

`columns` setting is fully usable now used for custom drawer items' grid customization. 

Personal settings is now a bit hidden for common users. It is removed from the user menu dropdown but can be reached with using the url https://admin.sensenet.com/personal-settings. [#707](https://github.com/SenseNet/sn-client/pull/707)

### @sensenet/client-core@3.1.0
- Connected to the permission API documentation and updates we ran into a couple of bugs and missing stuff in the core layer as well. The following fixes were made in the core package:
   - `setPermissionInheritance` method is removed and possibility of break the inheritance is moved to the method `setPermissions` as an optional param of the `PermissionRequestBody`.

>    - In permission query methods the param names where not the ones that should be used in the actual OData requests so they are changed: `kind` -> `indentityKind`; `level` -> `permissionLevel`; `member` -> `memberPath`

Reviewer: ```suggestion
   - In permission query methods the param names were not the ones that should be used in the actual OData requests so they are changed: `kind` -> `indentityKind`; `level` -> `permissionLevel`; `member` -> `memberPath`
```

[#721](https://github.com/SenseNet/sn-client/pull/721)

### @sensenet/redux@5.6.0
- As we move forward with the new documentation we improve the current client libraries with missing APIs or fixes. This time the sn-redux package got a bigger update. Permission related actions are added (HasPermission, GetPermissions, SetPermissions and for the permission queries), for further information and examples see the [API docs](https://docs.sensenet.com/api-docs/permissions). [#721](https://github.com/SenseNet/sn-client/pull/721)

### @sensenet/docviewer@2.2.0

> - As it is mentioned above the docviewer toolbar on the admin surface is now a bit simplified. This does not mean that you have to use it this way if you are using viewer as a standalone component. Related to the admin-ui changes minor improvement was also made on the docviewer toolbar widgets as well making rotation widgets configurable. You can now set how many rotation buttons you want to see setting the widgets `mode` to *clockwise*, *anticlockwise* or *all*. [#702](https://github.com/SenseNet/sn-client/pull/702)

Reviewer: ```suggestion
- As it is mentioned above the docviewer toolbar on the admin surface is now a bit simplified. This does not mean that you have to use it this way if you are using viewer as a standalone component. Related to the admin-ui changes minor improvement was also made on the docviewer toolbar widgets as well making rotation widgets configurable. You can now set how many rotation buttons you want to see by setting the widgets `mode` to *clockwise*, *anticlockwise* or *all*. [#702](https://github.com/SenseNet/sn-client/pull/702)
```

### @sensenet/picker@1.3.0

> - The above mentioned protected root logis is added to the pickers as well. So now the root of a picker is the same as the root of menu what you're currently working. [#707](https://github.com/SenseNet/sn-client/pull/707)

Reviewer 2: ```suggestion
- The above mentioned protected root logic is added to the pickers as well. So now the root of a picker is the same as the root of menu what you're currently working. [#707](https://github.com/SenseNet/sn-client/pull/707)
```

### Example apps

> - We've changed the Identity Server that is connected to the demo repository (dev.demo.sensenet.com) to new one that is for demo purposes and because of this we also had to change this hardcoded urls in the example apps. If you're using some of the example apps some way, take care of this changes too because the demo users can be logged in only through this demo is (its url is is.demo.sensenet.com).

Reviewer: ```suggestion
- We've changed the Identity Server that is connected to the demo repository (dev.demo.sensenet.com) to new one that is for demo purposes and because of this we also had to change this hardcoded urls in the example apps. If you're using some of the example apps some way, take care of this changes too because the demo users can be logged in only through this demo IS (its url is is.demo.sensenet.com).
```
